### PR TITLE
Improve the implementation of list multipart uploads

### PR DESCRIPTION
**Pull request by karoosnip**

### CODE REVIEW

File: src/main/java/com/qingstor/sdk/utils/QSStringUtil.java [hunk 1]
```diff
@@ -21,13 +21,21 @@
...
+import java.nio.charset.CharsetEncoder;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-import java.util.logging.Level;
 
 /** Created by on 11/4/15. */
 public class QSStringUtil {
+	
+	static CharsetEncoder asciiEncoder =
```
Comment (prettyxw) [inline src/main/java/com/qingstor/sdk/utils/QSStringUtil.java]: Strange code indent.

File: tests/scenario_impl/BucketTest.java [hunk 2]
```diff
@@ -94,7 +96,7 @@ public void initialize_the_bucket_without_zone() throws Throwable {
 	public void list_objects() throws Throwable {
 		// Write code here that turns the phrase above into concrete actions
 		Bucket.ListObjectsInput input = new Bucket.ListObjectsInput();
-		input.setLimit(20);
+		input.setLimit(20l);
```
Comment (prettyxw) [inline tests/scenario_impl/BucketTest.java]: What is `20l`?

File: tests/scenario_impl/BucketTest.java [hunk 3]
```diff
@@ -179,5 +181,25 @@ public void get_bucket_statistics_status_is(String arg1) throws Throwable {
 		System.out.println("get_bucket_statistics_status_msg:" + getBucketStatisticsOutput.getStatus());
 
 	}
+	
+	@Given("^an object created by Initiate Multipart Upload$")
+	public void an_object_created_by_Initiate_Multipart_Upload() throws Throwable {
+	    // Write code here that turns the phrase above into concrete actions
+		Bucket.InitiateMultipartUploadInput input = new Bucket.InitiateMultipartUploadInput();
+        Bucket.InitiateMultipartUploadOutput output = Bucket.initiateMultipartUpload("dididtes单独",input);
```
Comment (prettyxw) [inline tests/scenario_impl/BucketTest.java]: Code indent

File: tests/scenario_impl/BucketTest.java [hunk 4]
```diff
@@ -179,5 +181,25 @@ public void get_bucket_statistics_status_is(String arg1) throws Throwable {
...
+	    // Write code here that turns the phrase above into concrete actions
+		listMultipartUploadsOutput = Bucket.listMultipartUploads(null);
+        System.out.println(listMultipartUploadsOutput.getMessage());
+	}
+
+	@Then("^list multipart uploads count is (\\d+)$")
+	public void list_multipart_uploads_count_is(int arg1) throws Throwable {
+	    // Write code here that turns the phrase above into concrete actions
+		System.out.println(listMultipartUploadsOutput.getUploads().size());
+	}
```
Comment (prettyxw) [inline tests/scenario_impl/BucketTest.java]: Please abort the initiated multipart upload previously.

File: tests/scenario_impl/ObjectMultiTest.java [hunk 5]
```diff
@@ -95,8 +95,8 @@ public void initiate_multipart_upload_status_code_is(int arg1) throws Throwable
         TestUtil.assertEqual(initOutput.getStatueCode(),arg1);
     }
 
-    @When("^upload the first part$")
-    public void upload_the_first_part() throws Throwable {
+    @When("^upload the first part with key \"([^\"]*)\"$")
+    public void upload_the_first_part(String arg1) throws Throwable {
```
Comment (prettyxw) [inline tests/scenario_impl/ObjectMultiTest.java]: Rename `arg1` to `objectKey` will be much better.

File: tests/scenario_impl/ObjectTest.java [hunk 6]
```diff
@@ -33,7 +33,7 @@
 public class ObjectTest {
 
 
-    private static Bucket subService;
+	private static Bucket subService;
     private static String bucketName = TestUtil.getBucketName();
     private static String chinesePrefix = "中文测试/";
     private static String chineseSuffix = "/中 文test";
```
Comment (prettyxw) [inline tests/scenario_impl/ObjectTest.java]: Still need the `chinesePrefix` and `chineseSuffix`?

File: tests/scenario_impl/BucketTest.java (hunk 2 above)
Comment (prettyxw) [inline tests/scenario_impl/BucketTest.java]: Got it.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,29 @@
...
+
+- Url inputstream buffer read.
+
+## [v2.1.3] - 2017-01-23
+
+### Fixed
+
+- Assii character encode.
+
+
```
Comment (Xuanwo) [inline CHANGELOG.md]: This empty line should be removed.

File: CHANGELOG.md [hunk 8]
```diff
@@ -39,8 +62,9 @@ All notable changes to QingStor SDK for JAVA will be documented in this file.
 
 - QingStor SDK for the JAVA programming language.
 
+[v2.1.5]: https://github.com/yunify/qingstor-sdk-java/compare/2.1.4...2.1.5
+[v2.1.4]: https://github.com/yunify/qingstor-sdk-java/compare/2.1.3...2.1.4
+[v2.1.3]: https://github.com/yunify/qingstor-sdk-java/compare/2.1.2...2.1.3
 [v2.1.2]: https://github.com/yunify/qingstor-sdk-java/compare/2.1.1...2.1.2
 [v2.1.1]: https://github.com/yunify/qingstor-sdk-java/compare/2.1.0...2.1.1
-[v2.1.0]: https://github.com/yunify/qingstor-sdk-java/compare/2.0.1...2.1.0
```
Comment (Xuanwo) [inline CHANGELOG.md]: Why removed?

File: tests/scenario_impl/ObjectTest.java [hunk 9]
```diff
@@ -33,10 +33,9 @@
 public class ObjectTest {
 
 
-    private static Bucket subService;
+	private static Bucket subService;
```
Comment (Xuanwo) [inline tests/scenario_impl/ObjectTest.java]: strange indent.

File: tests/scenario_impl/ObjectTest.java (hunk 6 above)
Comment (karoosnip) [inline tests/scenario_impl/ObjectTest.java]: Removed

File: tests/scenario_impl/ObjectMultiTest.java (hunk 5 above)
Comment (karoosnip) [inline tests/scenario_impl/ObjectMultiTest.java]: Fixed

File: tests/scenario_impl/BucketTest.java (hunk 4 above)
Comment (karoosnip) [inline tests/scenario_impl/BucketTest.java]: Add  assert equal judgment

File: tests/scenario_impl/BucketTest.java (hunk 3 above)
Comment (karoosnip) [inline tests/scenario_impl/BucketTest.java]: Fixed

File: tests/scenario_impl/ObjectTest.java (hunk 9 above)
Comment (karoosnip) [inline tests/scenario_impl/ObjectTest.java]: Fixed

**Pull request merged**